BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -366,6 +378,12 @@ def override_process_line_with_version2(self, thebuilder, filelist, OverrideEntr
             return result
         # END: override_process_line_version2(self, thebuilder, filelist, OverrideEntry, m_node, status)
 
+        def deprecation_process_line_with_version1(self, thebuilder, filepath, DeprecationEntry):
+            if thebuilder.env.GetValue("ALLOW_DEPRECATED_MODULES") == "TRUE":

```

Comment:
I would lean more towards having this always on, and allowing a platform to add specific modules to be ignored (but still not disable the entire deprecation check).

something along the lines (but you don't need to do it exactly like this:

``` python
if filepath in thebuilder.env.GetValue("IGNORE_DEPRECATED_MODULES", "").split(";"):
    return self.OverrideResult.OR_ALL_GOOD
```

Then the platform can set it via:

```python
self.env.SetValue("IGNORE_DEPRECATED_MODULES", "MyPkg/Library/MyLib1/MyLib1.inf;MyPkg/Library/MyLib2/MyLib2.inf", "Platform Hardcoded")
```

Not sure if the filepath at this point is edk2-package path relative, or absolute, so may take some testing.